### PR TITLE
chore: [VUMM-758] Do not use personal workspace for permission v2

### PR DESCRIPTION
**Pull request by evgeniilezhnin**
## Impact and Context

The concept of a "personal workspace" is being removed in permission v2. This PR removes `Connection.get_personal_workspace()`.

## Risks and Area of Effect

This should have no impact for users. New clients can continue to be used with pre-permission v2 platforms.

`Connection.get_personal_workspace()` was only being used as a fallback for `Connection.get_default_workspace()` for backends older than Jan 2021 (https://github.com/VertaAI/modeldb/pull/1915), and all users are assigned a `default_workspace_id` as soon as they are created ([Slack thread](https://vertaai.slack.com/archives/C0306V04EJC/p1674757850181529)).

There is an issue where internal accounts bootstrapped with UAC aren't assigned a `default_workspace_id`, but that's a separate matter for the backend, and can be worked around clientside.

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

Test suite passes against a v1 environment ([Slack thread](https://vertaai.slack.com/archives/C03PWNZU5CM/p1674775415283039?thread_ts=1674666909.564999&cid=C03PWNZU5CM)).

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Revert this PR.

### CODE REVIEW

**Comment by liuverta** [comment]
~fyi all: Please don't merge this until Wednesday-ish! I need to publish a new version first before we merge a bunch of these behavior-breaking changes 😭~

☝️ This is done.

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]: Honestly, I think we could just remove this entire `if-else`. And instead, we can simply `return self.get_workspace_name_from_id(workspace_id)`.

The `else:  # old backend` is for before January 2020, so I'd say it's reasonable to remove. And then we can also remove `def get_personal_workspace()`

File: client/verta/verta/client.py
Comment (liuverta) [inline client/verta/verta/client.py:379]: Good catch! 👍

**Review by liuverta — CHANGES_REQUESTED** [review]

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py]: removed

**Review by liuverta — APPROVED** [review]

**Review by liuverta — APPROVED** [review]
Looks good to merge!

**Comment by evgeniilezhnin** [comment]
will wait some time until permission v2 flag enabled on test environments before merging this PR

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]:
```suggestion
        else:
```

Would you mind also removing this comment, for clarity?

**Review by liuverta — APPROVED** [review]

**Pull request merged**
